user_signup/main.py

```diff
@@ -7,18 +7,72 @@
 	loader=jinja2.FileSystemLoader(os.path.dirname(__file__)), 
 	autoescape=True)
 
-USER_RE = re.compile(r"^[a-zA-Z0-9_-]{3,20}$")
-PW_RE = re.compile(r"^.{3,20}$")
-EMAIL_RE = re.compile(r"^[\S]+@[\S]+.[\S]+$")
-
-def validate_username(username):
-	return USER_RE.match(username)
-
-def validate_password(password):
-	return PW_RE.match(password)
-
-def validate_email(email):
-	return EMAIL_RE.match(email)
+USER_RE = r"^[a-zA-Z0-9_-]{3,20}$"
+PW_RE = r"^.{3,20}$"
+EMAIL_RE = r"^[\S]+@[\S]+.[\S]+$"
+
+USER_ERROR = "Invalid username"
+PW_ERROR = "Invalid password"
+VERIFY_ERROR = "Passwords don't match"
+EMAIL_ERROR = "Invalid email address"
+
+
+def build_validation_re(USER_RE, PW_RE, EMAIL_RE):
+	return {'username': re.compile(USER_RE),
+		'password': re.compile(PW_RE),
+		'email': re.compile(EMAIL_RE)}
+
+
+def build_error_messages(USER_ERROR, PW_ERROR, VERIFY_ERROR, EMAIL_ERROR):
+	return {'username': USER_ERROR,
+		'password': PW_ERROR,
+		'verify': VERIFY_ERROR,
+		'email': EMAIL_ERROR}
+
+
+def validate(form_input, input_type):
+	validation_re = build_validation_re(USER_RE, PW_RE, EMAIL_RE)
+	validation_re = validation_re[input_type]
+	return validation_re.match(form_input)
```

Comment:
As I mentioned above, then this could be just:
```python
def validate(form_input, input_type):
  validation_re = VALIDATION_MATCHERS[input_type]
  return validation_re.match(form_input)
```